### ISSUE TITLE
Developer Tools cannot show Okhttp3 network response

### ISSUE THREAD

**Issue by ericksli** [facebook/stetho#409]
I am using Okhttp v.3.4.1 with Stetho v.1.3.1 and stetho-okhttp3 v.1.3.1.
I found that the developer tools cannot show the network response and there is a line *Provisional headers are shown* in request header section.

![screenshot of developer tools](http://i.imgur.com/ajwpxfX.png)

Here is how the request is made.

````java
try {
    OkHttpClient.Builder builder = new OkHttpClient.Builder();
    builder.followSslRedirects(false)
            .followRedirects(false)
            .cookieJar(new CookieJar() {
                private final HashMap<HttpUrl, List<Cookie>> cookieStore = new HashMap<>();

                @Override
                public void saveFromResponse(HttpUrl url, List<Cookie> cookies) {
                    cookieStore.put(url, cookies);
                }

                @Override
                public List<Cookie> loadForRequest(HttpUrl url) {
                    List<Cookie> cookies = cookieStore.get(url);
                    return cookies != null ? cookies : new ArrayList<Cookie>();
                }
            })
            .addNetworkInterceptor(new UserAgentInterceptor("Mozilla/5.0 (Linux; Android 6.0.1; D6503 Build/23.5.A.0.575) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.81 Mobile Safari/537.36"))
            .addNetworkInterceptor(new StethoInterceptor());
    OkHttpClient client = builder.build();

    Request probeRequest = new Request.Builder()
            .url("https://www.google.com.hk/?gws_rd=ssl")
            .build();

    Response probeResponse = client.newCall(probeRequest).execute();
    probeResponse.close();
} catch (IOException e) {
    Log.e(TAG, "IOException", e);
}
````

**Comment by krissrex** [comment]
It's because of a `NullPointerException` in `StethoInterceptor$OkHttpInspectorResponse#connectionId`.

I believe this is the cause:
`OkHttpClient#newCall` returns a `RealCall`, where the `Chain` becomes a `RealCall$ApplicationInterceptorChain`. The problem is caused by `RealCall$ApplicationInterceptorChain#connection`, as it always returns null.
Stetho calls `mConnection.hashCode()` on this `null` connection, and the call fails.

I hotfixed it in my app by using a static `AtomicInteger` with `getAndIncrement` instead of `mConnection.hashCode()`. However, I don't know if this satisfies the requirements for a connection ID. Anyways, I can see the response now, so it seems to be working.

If anyone is wondering, I'm using these dependencies:
```
'com.facebook.stetho:stetho-okhttp3:1.3.1'
'com.squareup.okhttp3:okhttp:3.2.0'
```` 

My (possibly incorrect) hotfix:
 (I copied the class `StethoInterceptor` into a package I created in my project with the same name)

```java
package com.facebook.stetho.okhttp3;

public class PatchedStethoInterceptor implements Interceptor {
    private static final AtomicInteger networkId = new AtomicInteger(0);

    ...

    private static class OkHttpInspectorResponse implements NetworkEventReporter.InspectorResponse {

        ...

        @Override
        public int connectionId() {
            return networkId.getAndIncrement(); //mConnection.hashCode();
        }

        ...

    }
    
    ...
}
```
